### PR TITLE
Implement Bearded Shame

### PR DESCRIPTION
**Pull request by victorphamdeveloper**
Feature: 
- Trim/shave/massage users. 
- Track who has trimmed/shaved/massaged. 
- Ability to create a new user.

Assumptions: 
- Beard length is 20 initially. 
- Every time a user trim, reduce beard length by 10. 
- If he shave, set it to 0.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,66 @@
...
+  # This function return all users who have shaved
+  def find_shave
+    @users = User.where(:shaved => true)
+    render 'list'
+  end 
+
+  # This function return all users who have trim
+  # Initial beard length is init to 20. 
+  # So if beard length is smaller than 20 and not 0. 
+  # He must have trimmed
```
Comment (nicosuria) [inline app/controllers/users_controller.rb]: I like this logic.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,7 +1,66 @@
 class UsersController < ApplicationController
 
+  require 'beard_trimmer'
+  require 'beard_trimmer/with_massage'
   def list
     @users = User.all
   end
+  
+  # This function return all users who have massaged
+  def find_massage
```
Comment (nicosuria) [inline app/controllers/users_controller.rb]: Are you familiar with REST?

Comment (nicosuria) [inline app/controllers/users_controller.rb]: Would have liked to see this moved into the `update` action, with 'massage', 'trim' and 'shave' just be parameters in the `put` request.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,7 +1,66 @@
...
+    @user.beard_length = 20
+    @user.last_massage_type = nil
+    if @user.save
+      redirect_to action: "list"
+    else
+      render 'new'
+    end
+  end
+
+  def user_params
```
Comment (nicosuria) [inline app/controllers/users_controller.rb]: As a best practice for controllers, if it's not a routable action make it `private`

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,7 +1,66 @@
 class UsersController < ApplicationController
 
+  require 'beard_trimmer'
+  require 'beard_trimmer/with_massage'
```
Comment (more-ron) [inline app/controllers/users_controller.rb]: What if you only want the massage?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (more-ron) [inline app/controllers/users_controller.rb]: What if the guy has 19 meters of beard?

**Comment by vincentpaca** [comment]
spec?

**Comment by vincentpaca** [comment]
Closing this PR.

**Pull request closed without merging**